usersimcrs/core/information_need.py

```diff
@@ -0,0 +1,94 @@
+"""Interface to represent an information need.
+
+The information need comprises three elements: constraints, requests, and target
+items. The constraints specify the slot-value pairs that the item of interest
+must satisfy, while the requests specify the slots for which the user wants
+information. The target items represent the "ground truth" items that the user
+is interested in.
+"""
+
+from __future__ import annotations
+
+import random
+from collections import defaultdict
+from typing import Any, Dict, List
+
+from usersimcrs.core.simulation_domain import SimulationDomain
+from usersimcrs.items.item import Item
+from usersimcrs.items.item_collection import ItemCollection
+
+
+def generate_random_information_need(
+    domain: SimulationDomain, item_collection: ItemCollection
+) -> InformationNeed:
+    """Generates a random information need based on the domain.
+
+    It randomly selects one target item and sets constraints and requests slots.
+    The value of constraints are derived from the target's properties. The
+    number of constraints and requests are also randomly determined.
+
+    Args:
+        domain: Domain knowledge.
+        item_collection: Collection of items.
+
+    Returns:
+        Information need.
+    """
+    target_item = random.choice(list(item_collection._items.values()))
+
+    constraints = {}
+    informable_slot = set(domain.get_informable_slots()).intersection(
+        target_item.properties.keys()
+    )
+    nb_constraints = random.randint(1, len(informable_slot))
```

Comment:
Nit: `nb_` is not necessarily obvious, how about replacing with `num_`? Also for `nb_requests`.

Reply:
Done.